### PR TITLE
Add TransformerCriterion

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

Add TransformerCriterion:

The criterion that takes two modules to transform input and target, and take
one criterion to compute the loss with the transformed input and target.

## How was this patch tested?

unit test

### CODE REVIEW

**Comment by yangw1234** [comment]
pr validation pass

File: docs/docs/APIGuide/Losses.md [hunk 1]
```diff
@@ -2186,3 +2186,62 @@ loss = criterion.forward(input, target)
...
+**Scala:**
+```scala
+val criterion = TransformerCriterion(criterion, Some(inputTransformer), Some(targetTransformer))
+```
+**Python:**
+```python
+criterion = TransformerCriterion(criterion, input_transformer, targetTransformer)
+```
+
+The criterion that takes two modules (optional) to transform input and target, and take
```
Comment (shane-huang) [inline docs/docs/APIGuide/Losses.md]: Elaborate the purpose of this criterion in doc. e.g. "This criterion is used for ..." or in what situations.

Comment (yangw1234) [reply docs/docs/APIGuide/Losses.md]: @shane-huang Added

**Comment by yangw1234** [comment]
pr validation pass
